x/ibc/04-channel/keeper/handshake_test.go

```diff
@@ -4,624 +4,634 @@ import (
 		}, false},
 		{"port capability not found", func() {
-			suite.chainA.CreateClient(suite.chainB)
-			suite.chainB.CreateClient(suite.chainA)
-			suite.chainA.createConnection(
-				testConnectionIDB, testConnectionIDA, testClientIDB, testClientIDA,
-				connection.OPEN,
-			)
-			suite.chainB.createConnection(
-				testConnectionIDA, testConnectionIDB, testClientIDA, testClientIDB, connection.OPEN)
-			suite.chainB.createChannel(testPort1, testChannel1, testPort2, testChannel2, types.INIT, types.ORDERED, testConnectionIDA)
+			_, _, connA, connB = suite.coordinator.SetupClientConnections(suite.chainA, suite.chainB, clientexported.Tendermint)
+			suite.coordinator.ChanOpenInit(suite.chainA, suite.chainB, connA, connB, types.ORDERED)
+
 			portCap = capabilitytypes.NewCapability(3)
 		}, false},
 	}
 
-	for i, tc := range testCases {
+	for _, tc := range testCases {
 		tc := tc
-		i := i
 		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
 			suite.SetupTest() // reset
-
-			var err error
-			portCap, err = suite.chainA.App.ScopedIBCKeeper.NewCapability(suite.chainA.GetContext(), host.PortPath(testPort2))
-			suite.Require().NoError(err, "could not create capability")
+			heightDiff = 0    // must be explicitly changed
 
 			tc.malleate()
+			counterparty := types.NewCounterparty(connA.FirstOrNextTestChannel().PortID, connA.FirstOrNextTestChannel().ID)
+			channelB := connB.FirstOrNextTestChannel()
 
-			suite.chainA.updateClient(suite.chainB)
-			suite.chainB.updateClient(suite.chainA)
-			proof, proofHeight := queryProof(suite.chainB, channelKey)
+			channelKey := host.KeyChannel(counterparty.PortID, counterparty.ChannelID)
+			proof, proofHeight := suite.chainA.QueryProof(channelKey)
+
+			cap, err := suite.chainB.App.IBCKeeper.ChannelKeeper.ChanOpenTry(
+				suite.chainB.GetContext(), types.ORDERED, []string{connB.ID},
+				channelB.PortID, channelB.ID, portCap, counterparty, ibctesting.ChannelVersion, ibctesting.ChannelVersion,
+				proof, proofHeight+heightDiff,
+			)
 
 			if tc.expPass {
-				cap, err := suite.chainA.App.IBCKeeper.ChannelKeeper.ChanOpenTry(
-					suite.chainA.GetContext(), types.ORDERED, []string{testConnectionIDB},
-					testPort2, testChannel2, portCap, counterparty, testChannelVersion, testChannelVersion,
-					proof, proofHeight+1,
-				)
-				suite.Require().NoError(err, "valid test case %d failed: %s", i, tc.msg)
+				suite.Require().NoError(err)
 				suite.Require().NotNil(cap)
+
 				chanCap, ok := suite.chainA.App.ScopedIBCKeeper.GetCapability(
 					suite.chainA.GetContext(),
-					host.ChannelCapabilityPath(testPort2, testChannel2),
+					host.ChannelCapabilityPath(channelB.PortID, channelB.ID),
 				)
 				suite.Require().True(ok, "could not retrieve channel capapbility after successful ChanOpenInit")
-				suite.Require().Equal(chanCap.String(), cap.String(), "channel capability is not correct")
+				suite.Require().Equal(chanCap.Index, cap.Index, "channel capability is not correct")
```

Comment:
Simply printing the index isn't a good idea. Since the indices can be the same without actually having same memory location. Would recommend sticking with `String` in err msg for capabilities

Reply:
Hmm, using `String()` fails which is why I changed it. I'm not sure what I'm doing wrong through in handling chanCap?

Reply:
fixed it, issue was I forgot to update which chain to get the capability from